### ISSUE TITLE
Ошибка «Запись не найдена в менеджере имен базы данных»

### ISSUE THREAD

**Issue by SminexIT** [1C-Company/1c-edt-issues#92]
### Описание ошибки

ИБ клиент-серверная (MS SQL), примерный размер 120 ГБ

При длительном обновлении ИБ (длительной реструктуризации), процесс иногда падает с ошибкой ..DBNames. 

![edt_01](https://user-images.githubusercontent.com/94164898/141440632-60d76832-2a8c-4299-9f4e-d8098f35cf12.png)

Если после этого попробовать сразу снова обновить ИБ из EDT, возникает ошибка «Запись не найдена в менеджере имен базы данных», после чего ИБ не подлежит восстановлению (по крайней мере пока способы не найдены)

![edt_03](https://user-images.githubusercontent.com/94164898/141443009-a3ae6d8d-6ac7-4aeb-916a-b6e5af22fff2.png)

(Было предположение, что это как-то связано с "удалёнными" расширениями, но удаление всех расширений в базе и тестирование и исправление расширений на логическую целостность не помогает)

Логи EDT показывают следующую информацию: Таблица Chrc29665 отсутствует в схеме базы данных (pos=100), где Chrc — планы видов характеристик (в данном случае ошибка возникла после добавления предопределенного реквизита в План Видов Характеристик)

![edt_02](https://user-images.githubusercontent.com/94164898/141440675-09dd499e-9507-4776-b25f-4017c5f6db19.png)

Но при анализе таблиц БД, такую таблицу (Chrc29665) найти не удалось.

Ошибка повторялась как минимум:
- после добавления предопределенного реквизита в План Видов Характеристик
- после добавления реквизита в документ
- после изменения длины строкового реквизита в справочнике (в этом случае ошибка была: Таблица Reference28658 отсутствует в схеме базы данных (pos=1525), данная таблица в ИБ так же не найдена)

### Как воспроизвести

1. Добавить/Изменить любой реквизит, в любом объекте, в котором в БД содержится очень много элементов (чтобы вызвать длительную реструктуризацию), 
2. Кликнуть по разрабатываемому приложению - Обновить конфигурацию
3. Дождаться появления ошибки ..DBNames (понятно что появляется далеко не всегда, обычно при проблемах с сетью)
4. Повторно кликнуть по разрабатываемому приложению - Обновить конфигурацию

### Скриншоты

![edt_04](https://user-images.githubusercontent.com/94164898/141477222-4947dfaa-c790-46b1-a702-62827dd78d41.png)

### Ожидаемое поведение

ошибка «Запись не найдена в менеджере имен базы данных» не возникает и ИБ не разрушается

### Лог рабочей области

[trace.log](https://github.com/1C-Company/1c-edt-issues/files/7527953/trace.log)
[ATT35712.log](https://github.com/1C-Company/1c-edt-issues/files/7527954/ATT35712.log
[bak_0.log](https://github.com/1C-Company/1c-edt-issues/files/7528005/bak_0.log)
)
[trace.bak_0.log](https://github.com/1C-Company/1c-edt-issues/files/7528008/trace.bak_0.log)

### Версия 1С:EDT

2021.2.6

### Операционаня система

Windows

### Установленые плагины

Нет плагинов

### Дополнительная информация

Версия платформы 8.3.20.1549

**Comment by mrshadow300373** [comment]
> (Было предположение, что это как-то связано с "удалёнными" расширениями
У меня точно такая же проблема произошла, после удаления 5 из 6 расширений из базы. И есть BAK этой базы до удаления расширений 100% воспроизводимость, сейчас оформляю на v8@1c.ru . EDT тут ни причём совсем.

**Comment by MaksimDzyuba** [comment]
> > (Было предположение, что это как-то связано с "удалёнными" расширениями
> 
> У меня точно такая же проблема произошла, после удаления 5 из 6 расширений из базы. И есть BAK этой базы до удаления расширений 100% воспроизводимость, сейчас оформляю на [v8@1c.ru](mailto:v8@1c.ru) . EDT тут ни причём совсем.

Спасибо за информацию, если не трудно, можете приложить номер ошибки, которую зарегистрировали на платформу, чтобы мы отследили, что с ней, и эту ошибку закрыли, пока просто понижу статут критичности

**Comment by mrshadow300373** [comment]
> > > (Было предположение, что это как-то связано с "удалёнными" расширениями
> > 
> > 
> > У меня точно такая же проблема произошла, после удаления 5 из 6 расширений из базы. И есть BAK этой базы до удаления расширений 100% воспроизводимость, сейчас оформляю на [v8@1c.ru](mailto:v8@1c.ru) . EDT тут ни причём совсем.
> 
> Спасибо за информацию, если не трудно, можете приложить номер ошибки, которую зарегистрировали на платформу, чтобы мы отследили, что с ней, и эту ошибку закрыли, пока просто понижу статут критичности

Код ошибки они не прислали, номер обращения HL-500056
Причин тоже не сказали, предложили способ как "вылечить" проблему, но у меня согласно их инструкции не получилось. У меня при попытке добавить изменить структуру регистра, на который они сослались реорганизация ИБ не проходит. Текст переписки, в нём нет ничего конфиденциального, прилагаю.

[(#HL-500056) RE[8]_ После удаления расширений в работоспособной ИБ база перестаёт открываться в режиме 1С_Предприятия с ошибкой.zip](https://github.com/1C-Company/1c-edt-issues/files/9236505/HL-500056.RE.8._.1._.zip)

Отвечать я им более не стал, так как диалог мне напомнил разговор слепого с глухим. Я их спрашиваю про причины, они мне предлагают лекарство:

> Мне нужно понять
> что приводит к подобным ситуациям, какие мои действия были неправильные действия, почему расширения склеились.

Как это произошло сказать сложно. Восстановить сценарий поломки по сломанной когда-то, возможно уже достаточно давно, базы практически невозможно.

Можно предположить, что может помочь реструктуризация этого регистра. 
Открываем конфигурацию, модифицируем регистр (как угодно, добавить ресурс). Сохраняем. _<<--здесь у меня не получилось_
Проверяем ТИИ. Ошибок нет. 
Конфигурация работает. Можно удалить расширение IEX и все остальные.

**Comment by MaksimDzyuba** [comment]
да, посмотрел, не понятно, так как обращение Ваше уже закрыто, и я не могу понять делалось ли вообще что-то(
